nucleus/__init__.py

```diff
@@ -1285,15 +1296,88 @@ def create_dataset_from_dir(
 
         if len(items) == 0:
             print(f"Did not find any items in {dirname}")
-            return None
+            return existing_dataset
 
         if len(items) > GLOB_SIZE_THRESHOLD_CHECK and not skip_size_warning:
             raise Exception(
                 f"Found over {GLOB_SIZE_THRESHOLD_CHECK} items in {dirname}. If this is intended, set skip_size_warning=True when calling this function."
             )
 
-        dataset = self.create_dataset(
-            name=dataset_name, use_privacy_mode=use_privacy_mode
-        )
-        dataset.append(items, asynchronous=False)
+        if existing_dataset:
+            dataset = existing_dataset
+        else:
+            folder_name = os.path.basename(_dirname.rstrip("/"))
+            dataset_name = dataset_name or folder_name
+            dataset = self.create_dataset(
+                name=dataset_name, use_privacy_mode=use_privacy_mode
+            )
+        dataset.append(items, asynchronous=False, update=update_items)
         return dataset
+
+    def create_dataset_from_dir(
+        self,
+        dirname: str,
+        dataset_name: Optional[str] = None,
+        use_privacy_mode: bool = False,
+        privacy_mode_proxy: str = "",
+        allowed_file_types: Tuple[str, ...] = ("png", "jpg", "jpeg"),
+        skip_size_warning: bool = False,
+    ) -> Union[Dataset, None]:
+        """
+        Create a dataset by recursively crawling through a directory.
+        A DatasetItem will be created for each unique image found.
+
+        Parameters:
+            dirname: Where to look for image files, recursively
+            dataset_name: If none is given, the parent folder name is used
+            use_privacy_mode: Whether the dataset should be treated as privacy
+            privacy_mode_proxy: Endpoint that serves image files for privacy mode, ignore if not using privacy mode.
+                The proxy should work based on the relative path of the images in the directory.
+            allowed_file_types: Which file type extensions to search for, ie: ('jpg', 'png')
+            skip_size_warning: If False, it will throw an error if the script globs more than 500 images. This is a safety check in case the dirname has a typo, and grabs too much data.
+        """
+        return self._create_or_update_dataset_from_dir(
+            dirname,
+            dataset_name=dataset_name,
+            use_privacy_mode=use_privacy_mode,
+            privacy_mode_proxy=privacy_mode_proxy,
+            allowed_file_types=allowed_file_types,
+            skip_size_warning=skip_size_warning,
+        )
+
+    def update_dataset_from_dir(
```

Comment:
Maybe this one would be better to be had on the dataset class itself. I think it's a better workflow to create/fetch a dataset explicitly then ingesting items into it.

Reply:
Agreed, I think that's better encapsulation

Reply:
I think that's a good suggestion. It would avoid the user having to copy the dataset id.

I think we can even rename it to:

`dataset.add_images_from_dir(dirname)`

Reply:
Addressed